yt_dlp/extractor/instagram.py

```diff
@@ -234,7 +234,9 @@ def _real_extract(self, url):
                 media, lambda x: x['edge_media_to_caption']['edges'][0]['node']['text'],
                 compat_str) or media.get('caption')
             title = media.get('title')
-            thumbnail = media.get('display_src') or media.get('display_url')
+            display_resources = media.get('display_resources')
+            if not display_resources:
+                display_resources = [{'src': media.get('display_src') or media.get('display_url')}]
```

Comment:
Now that this is being done `if thumbnail.get('src')`
```suggestion
                display_resources = [{'src': media.get('display_src')}, {'src': media.get('display_url')}]
```
TBH i'd remove the ` if not display_resources:` and append anyways

something like 
```py
            display_resources = media.get('display_resources') or []
            display_resources.extend([{'src': media.get('display_src')}, {'src': media.get('display_url')}])
```